chore: Update README to highlight Comet benefits

andygrove opened the pull request:
## Which issue does this PR close?

<!--
We generally require a GitHub issue to be filed for all bug fixes and enhancements and this helps us generate change logs for our releases. You can link an issue to this PR using the GitHub syntax. For example `Closes #123` indicates that this PR will close issue #123.
-->

N/A

## Rationale for this change

<!--
 Why are you proposing this change? If this is already explained clearly in the issue then this section is not needed.
 Explaining clearly why changes are proposed helps reviewers understand your changes and offer better suggestions for fixes.  
-->

By focusing on the benefits of Comet, and showing some performance results, we can hopefully make the project more compelling to users who may consider trying it out.

## What changes are included in this PR?

<!--
There is no need to duplicate the description in the issue here but it is sometimes worth providing a summary of the individual changes in this PR.
-->

New README. You can see the rendered version [here](https://github.com/andygrove/datafusion-comet/tree/update-readme2)

## How are these changes tested?

<!--
We typically require tests for all PRs in order to:
1. Prevent the code from being accidentally broken by subsequent changes
2. Serve as another way to document the expected behavior of the code

If tests are not included in your PR, please explain why (for example, are they covered by existing tests)?
-->

kazuyukitanimura commented on README.md:
```diff
@@ -19,58 +19,84 @@ under the License.
 
-Linux, Apple OSX (Intel and M1)
+## Use Existing Hardware
 
-## Requirements
+Comet leverages your existing hardware infrastructure, eliminating the need for costly hardware upgrades or
+specialized hardware accelerators. By maximizing the utilization of commodity hardware, Comet ensures
+cost-effectiveness and scalability for your Spark deployments.
 
```
nit `existing hardware` is a bit ambiguous, I feel. Did you mean traditional CPU?

`eliminating the need for costly hardware upgrades` sounds like the assumption is other options require accelerators, not sure if it is always the case...

andygrove replied:
That's good feedback. Yes, I meant commodity hardware (CPUs) rather than needing GPU, FPGA, etc. I will try and refine this tomorrow.

andygrove replied:
I have updated this.

comphead commented on README.md:
```diff
@@ -19,58 +19,84 @@ under the License.
+When running DataFusion standalone (without Spark), the run time is 3.9x faster.
 
-<a href="docs/source/_static/images/comet-overview.png"><img src="docs/source/_static/images/comet-overview.png" align="center" height="600" width="750" ></a>
+Comet is not yet achieving full DataFusion speeds in all cases, but with future work we aim to provide a 2x-4x speedup 
```
if pure DF gives 3.9x, is that possible Comet built on top of DF to give 4x?

andygrove replied:
I was rounding up to the nearest whole percent here, but of course it will be challenging. DataFusion isn't performing shuffle operations, but I think that DataFusion performance sets a hard limit on what we can do with Comet, so 2-4x seems to be our expected range (unless there are future optimizations in DataFusion).

comphead commented on README.md:
```diff
@@ -19,58 +19,84 @@ under the License.
+When using Comet, the overall run time is reduced from 649 seconds to 440 seconds, which is 1.5x faster.
 
-The following diagram illustrates the architecture of Comet:
+When running DataFusion standalone (without Spark), the run time is 3.9x faster.
```
```suggestion
When running TPC-H queries with DataFusion standalone (without Spark), the overall  run time is 3.9x faster.
```

comphead replied:
We can probably do a table? 

| Spark | Comet | DataFusion
|-------|--------|--------------
|   649 | 440 | 180 |

comphead reviewed (approved):
lgtm thanks @andygrove

kazuyukitanimura commented on README.md:
```diff
@@ -19,58 +19,84 @@ under the License.
+for many use cases.
 
-## Current Status
+![](docs/source/_static/images/tpch_allqueries.png)
```
nit: should we mention the experimental environment such as AWS instance type, where the data is located, etc ...

kazuyukitanimura reviewed (approved):
All my comments are nits

andygrove commented:
Thanks for the reviews @viirya @kazuyukitanimura @comphead. I have addressed the feedback so will go ahead and merge this and we can follow up with future PRs if we want to refine this more.
